pkg/api/register.go

```diff
@@ -78,12 +78,12 @@ func init() {
 
 	// Register Unversioned types
 	// TODO this should not be done here
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.ListOptions{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.Status{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.APIVersions{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.APIGroupList{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.APIGroup{})
-	Scheme.AddKnownTypes(SchemeGroupVersion, &unversioned.APIResourceList{})
+	Scheme.AddKnownTypes(unversioned.SchemeGroupVersion, &unversioned.ListOptions{})
```

Comment:
> We can move the registrations to unversioned.SchemeGroupVersion to the pkg/api/unversioned/register.go, that will be logical, though that just duplicates the registration.

I tried to move the registration, but because the type is `unversioned.GroupVersion`, that creates a cycle (`api` needs `unversioned.GroupVersion`, `unversioned` needs `api.Scheme`).  I think we should consider moving the metadata structs (`GroupVersion`, `GroupVersionKind`, etc) to a different package, but I definitely don't want to do that here.

If you're not fully convinced, I'll revert this bit to the old code and open a smaller pull where we can talk about it.